src/dawn/SIR/SIR.h

```diff
@@ -363,8 +384,32 @@ struct SIR {
 
   /// @brief Dump SIR to the given stream
   friend std::ostream& operator<<(std::ostream& os, const SIR& Sir);
+
+  /// @brief Compares two SIRs for equality in contents
+  /// It is to note that the filenames are not compared here
+  /// and positions of expressions are omitted
+  bool operator==(const SIR& rhs) const;
+
+  /// @brief Compares two SIRs for inequality in contents
+  /// It is to note that the filenames are not compared here
+  /// and positions of expressions are omitted
+  bool operator!=(const SIR& rhs) const;
+
+  bool pointerMapComparison(const sir::GlobalVariableMap& map1,
+                            const sir::GlobalVariableMap& map2) const;
 };
 
+////////////// Should not be here.....
+/// @brief Compares the content of two shared pointers
+/// @param[in] shared pointer of type T
+/// @param[in] shared pointer of same type T
+/// @return true if contents of the shared pointers match (operator ==)
+template <typename T>
+bool sharedpointer_comparison(const std::shared_ptr<T>& comparate1,
```

Comment:
I think the name is misleading as it does not *actually* compare the pointer but rather the pointee. Maybe something like `pointeeComparison`? Also do you need to expose this function in the header? I think putting it in `SIR.cpp` would be more appropriate.